it seems it failed with 

```


Message: Could not resolve placeholder ${auto.test.resources.jpa.default.properties.hibernate.connection.url} |  
-- | --
Path Taken: SessionFactory.buildHibernateSessionFactoryBuilder(SessionFactoryBuilder sessionFactoryBuilder) --> SessionFactory.buildHibernateSessionFactoryBuilder([SessionFactoryBuilder sessionFactoryBuilder]) --> SessionFactoryBuilder.buildHibernateSessionFactoryBuilder(Metadata metadata,[JpaConfiguration jpaConfiguration]) |  
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: |  
Error instantiating bean of type  [org.hibernate.boot.SessionFactoryBuilder] |  
  |  
Message: Could not resolve placeholder ${auto.test.resources.jpa.default.properties.hibernate.connection.url} |  
Path Taken: SessionFactory.buildHibernateSessionFactoryBuilder(SessionFactoryBuilder sessionFactoryBuilder) --> SessionFactory.buildHibernateSessionFactoryBuilder([SessionFactoryBuilder sessionFactoryBuilder]) --> SessionFactoryBuilder.buildHibernateSessionFactoryBuilder(Metadata metadata,[JpaConfiguration jpaConfiguration]) |  
Caused by: io.micronaut.context.exceptions.ConfigurationException: Could not resolve placeholder ${auto.test.resources.jpa.default.properties.hibernate.connection.url} |  
``` 

--- a/starter-core/src/main/java/io/micronaut/starter/feature/database/SQLServer.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/database/SQLServer.java
@@ -16,7 +16,6 @@
 package io.micronaut.starter.feature.database;
 
 import io.micronaut.core.annotation.NonNull;
-import io.micronaut.core.util.StringUtils;
 import io.micronaut.starter.application.generator.GeneratorContext;
 import io.micronaut.starter.build.dependencies.Dependency;
 import io.micronaut.starter.feature.database.jdbc.JdbcFeature;
@@ -24,8 +23,6 @@ import io.micronaut.starter.feature.testresources.DbType;
 import io.micronaut.starter.feature.testresources.TestResources;
 import jakarta.inject.Singleton;
 
-import java.util.Collections;
-import java.util.Map;
 import java.util.Optional;
 
 import static io.micronaut.starter.feature.database.DataHibernateReactive.IO_VERTX_DEPENDENCY_GROUP;
@@ -135,9 +132,14 @@ public class SQLServer extends DatabaseDriverFeature {
     }
 
     @Override
-    public Map<String, Object> getAdditionalConfig(GeneratorContext generatorContext) {
-        return generatorContext.hasFeature(TestResources.class) ?
-                Collections.singletonMap("test-resources.containers.mssql.accept-license", StringUtils.TRUE) :
-                Collections.emptyMap();
+    public void apply(GeneratorContext generatorContext) {
+        super.apply(generatorContext);
+        if (generatorContext.hasFeature(TestResources.class)) {
+            generatorContext.getConfiguration().put("test-resources.containers.mssql.accept-license", acceptLicense());
+        }
+    }
+
+    protected boolean acceptLicense() {
+        return false;
     }
 }

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/database/SQLServerSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/database/SQLServerSpec.groovy
@@ -5,10 +5,11 @@ import io.micronaut.starter.BuildBuilder
 import io.micronaut.starter.build.BuildTestUtil
 import io.micronaut.starter.build.BuildTestVerifier
 import io.micronaut.starter.build.dependencies.Scope
-import io.micronaut.starter.feature.Feature
 import io.micronaut.starter.fixture.CommandOutputFixture
 import io.micronaut.starter.options.BuildTool
 import io.micronaut.starter.options.Language
+import spock.lang.IgnoreIf
+import spock.lang.Requires
 import spock.lang.Unroll
 
 class SQLServerSpec extends ApplicationContextSpec implements CommandOutputFixture {
@@ -40,6 +41,7 @@ class SQLServerSpec extends ApplicationContextSpec implements CommandOutputFixtu
         output["src/main/resources/application.properties"].contains("test-resources.containers.mssql.accept-license=false\n")
     }
 
+    @Requires({ jvm.current.isJava11Compatible() })
     void 'test for test-resources the accept-license property is set to false with hibernate reactive=#hibernateReactiveFeature'(
             String hibernateReactiveFeature) {
         when:

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/database/SQLServerSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/database/SQLServerSpec.groovy
@@ -5,6 +5,7 @@ import io.micronaut.starter.BuildBuilder
 import io.micronaut.starter.build.BuildTestUtil
 import io.micronaut.starter.build.BuildTestVerifier
 import io.micronaut.starter.build.dependencies.Scope
+import io.micronaut.starter.feature.Feature
 import io.micronaut.starter.fixture.CommandOutputFixture
 import io.micronaut.starter.options.BuildTool
 import io.micronaut.starter.options.Language
@@ -31,11 +32,23 @@ class SQLServerSpec extends ApplicationContextSpec implements CommandOutputFixtu
         [language, buildTool] << [Language.values().toList(), BuildTool.values()].combinations()
     }
 
-    void 'test for test-resources the accept-license property is set to true'() {
+    void 'test for test-resources the accept-license property is set to false'() {
         when:
         Map<String, String> output = generate(["sqlserver", "properties"])
 
         then:
-        output["src/main/resources/application.properties"].contains("test-resources.containers.mssql.accept-license=true\n")
+        output["src/main/resources/application.properties"].contains("test-resources.containers.mssql.accept-license=false\n")
+    }
+
+    void 'test for test-resources the accept-license property is set to false with hibernate reactive=#hibernateReactiveFeature'(
+            String hibernateReactiveFeature) {
+        when:
+        Map<String, String> output = generate(["sqlserver", "properties", hibernateReactiveFeature])
+
+        then:
+        output["src/main/resources/application.properties"].contains("test-resources.containers.mssql.accept-license=false\n")
+
+        where:
+        hibernateReactiveFeature << [HibernateReactiveJpa.NAME, DataHibernateReactive.NAME]
     }
 }

--- a/test-features/src/test/groovy/io/micronaut/starter/core/test/feature/database/AcceptLicenseSqlServer.groovy
+++ b/test-features/src/test/groovy/io/micronaut/starter/core/test/feature/database/AcceptLicenseSqlServer.groovy
@@ -1,0 +1,27 @@
+package io.micronaut.starter.core.test.feature.database
+
+import io.micronaut.context.annotation.Replaces
+import io.micronaut.starter.feature.database.SQLServer
+import io.micronaut.starter.feature.database.TestContainers
+import io.micronaut.starter.feature.database.jdbc.JdbcFeature
+import io.micronaut.starter.feature.testresources.TestResources
+import jakarta.inject.Singleton
+
+@Replaces(SQLServer.class)
+@Singleton
+class AcceptLicenseSqlServer extends SQLServer {
+
+    AcceptLicenseSqlServer(JdbcFeature jdbcFeature, TestContainers testContainers, TestResources testResources) {
+        super(jdbcFeature, testContainers, testResources)
+    }
+
+    @Override
+    boolean isVisible() {
+        return false
+    }
+
+    @Override
+    protected boolean acceptLicense() {
+        return true
+    }
+}

--- a/test-features/src/test/groovy/io/micronaut/starter/core/test/feature/database/HibernateReactiveJpaSpec.groovy
+++ b/test-features/src/test/groovy/io/micronaut/starter/core/test/feature/database/HibernateReactiveJpaSpec.groovy
@@ -1,5 +1,6 @@
 package io.micronaut.starter.core.test.feature.database
 
+import io.micronaut.starter.core.test.feature.database.templates.book
 import io.micronaut.starter.feature.database.HibernateReactiveJpa
 import io.micronaut.starter.feature.database.MariaDB
 import io.micronaut.starter.feature.database.MySQL
@@ -13,7 +14,6 @@ import io.micronaut.starter.options.Language
 import io.micronaut.starter.template.RockerWritable
 import io.micronaut.starter.test.CommandSpec
 import org.gradle.testkit.runner.BuildResult
-import io.micronaut.starter.core.test.feature.database.templates.book
 import spock.lang.Requires
 
 @Requires({ jvm.current.isJava11Compatible() })

